hazelcast/src/main/java/com/hazelcast/config/SqlConfig.java

```diff
@@ -0,0 +1,154 @@
+        return executorPoolSize;
+    }
+
+    /**
+     * Sets the number of threads responsible for query execution.
+     * <p>
+     * Normally the value of this parameter should be equal to the number of CPU cores.
+     * <p>
+     * Setting the value to less than the number of CPU cores will limit the degree of parallelism of the SQL subsystem. This
+     * may be beneficial if you would like to prioritize other CPU-intensive workloads on the same machine.
+     * <p>
+     * It is not recommended to set the value of this parameter greater than the number of CPU cores because it may decrease
+     * the system's overall performance due to excessive context switches.
+     * <p>
+     * Defaults to {@link #DEFAULT_EXECUTOR_POOL_SIZE}.
+     *
+     * @param executorPoolSize Number of threads responsible for query execution.
+     * @return This instance for chaining.
+     */
+    public SqlConfig setExecutorPoolSize(int executorPoolSize) {
+        checkPositive(executorPoolSize, "Executor pool size should be positive");
+
+        this.executorPoolSize = executorPoolSize;
+
+        return this;
+    }
+
+    /**
+     * Gets the number of threads responsible for network operations processing.
+     *
+     * @return Number of threads responsible for network operations processing.
+     */
+    public int getOperationPoolSize() {
+        return operationPoolSize;
+    }
+
+    /**
+     * Sets the number of threads responsible for network operations processing.
+     * <p>
+     * When Hazelcast members execute a query, they send commands to each other over the network to coordinate the execution.
+     * This includes requests to start or stop query execution, or a request to process a batch of data. These commands are
+     * processed in a separate operation thread pool, to avoid frequent interruption of running query fragments.
+     * <p>
+     * The default value should be good enough for the most workloads. You may want to increase the default value if you run
+     * very small queries, or the machine has a big number of CPU cores.
+     * <p>
+     * It is not recommended to set the value of this parameter greater than the number of CPU cores because it may decrease
+     * the system's overall performance due to excessive context switches.
+     *
+     * <p>
+     * Defaults to {@link #DEFAULT_OPERATION_POOL_SIZE}.
+     *
+     * @param operationPoolSize Number of threads responsible for network operations processing.
+     * @return This instance for chaining.
+     */
+    public SqlConfig setOperationPoolSize(int operationPoolSize) {
+        checkPositive(operationPoolSize, "Operation pool size should be positive");
+
+        this.operationPoolSize = operationPoolSize;
+
+        return this;
+    }
+
+    /**
+     * Gets the timeout in milliseconds that is applied to queries without an explicit timeout.
+     *
+     * @return Timeout in milliseconds.
+     */
+    public long getQueryTimeout() {
```

Comment:
I think it's a good convention to use something like `queryTimeoutMillis` because if you're editing the config file, the unit is not  always obvious. I think this convention is followed in rest of the config.

Reply:
Applied the suggestion.